test/node/node_test.cpp

```diff
@@ -239,6 +267,272 @@ TEST(NodeTest, StdMap) {
+
+  Node node;
+  node["primes"] = primes;
+  const auto actualPrimes = node["primes"].as<std::set<int>>();
+  EXPECT_EQ(primes, actualPrimes);
+}
+
+TEST(NodeTest, StdUnorderedSet) {
+  std::unordered_set<int> primes;
+  primes.insert(0);
+  primes.insert(1);
+  primes.insert(4);
+  primes.insert(9);
+  primes.insert(16);
+
+  Node node;
+  node["primes"] = primes;
+  const auto actualPrimes = node["primes"].as<std::unordered_set<int>>();
+  EXPECT_EQ(primes, actualPrimes);
+}
+
+TEST(NodeTest, StdBitset) {
+  std::bitset<8> bits;
+  bits.set(0, true);
+  bits.set(1, true);
+  bits.set(2, false);
+  bits.set(3, true);
+  bits.set(4, true);
+  bits.set(5, false);
+  bits.set(6, false);
+  bits.set(7, false);
+
+  Node node;
+  node["bits"] = bits;
+  const auto actual = node["bits"].as<std::bitset<8>>();
+  EXPECT_EQ(bits, actual);
+}
+
+TEST(NodeTest, StdBitsetWrongSize) {
+  std::bitset<8> bits;
+  bits.set(0, true);
+  bits.set(1, true);
+  bits.set(2, false);
+  bits.set(3, true);
+  bits.set(4, true);
+  bits.set(5, false);
+  bits.set(6, false);
+  bits.set(7, false);
+
+  Node node;
+  node["bits"] = bits;
+  EXPECT_THROW_REPRESENTATION_EXCEPTION((node["bits"].as<std::bitset<1>>()),
+                                        ErrorMsg::BAD_CONVERSION);
+  EXPECT_THROW_REPRESENTATION_EXCEPTION((node["bits"].as<std::bitset<2>>()),
+                                        ErrorMsg::BAD_CONVERSION);
+  EXPECT_THROW_REPRESENTATION_EXCEPTION((node["bits"].as<std::bitset<3>>()),
+                                        ErrorMsg::BAD_CONVERSION);
+  EXPECT_THROW_REPRESENTATION_EXCEPTION((node["bits"].as<std::bitset<4>>()),
+                                        ErrorMsg::BAD_CONVERSION);
+  EXPECT_THROW_REPRESENTATION_EXCEPTION((node["bits"].as<std::bitset<5>>()),
+                                        ErrorMsg::BAD_CONVERSION);
+  EXPECT_THROW_REPRESENTATION_EXCEPTION((node["bits"].as<std::bitset<6>>()),
+                                        ErrorMsg::BAD_CONVERSION);
+  EXPECT_THROW_REPRESENTATION_EXCEPTION((node["bits"].as<std::bitset<7>>()),
+                                        ErrorMsg::BAD_CONVERSION);
+  EXPECT_THROW_REPRESENTATION_EXCEPTION((node["bits"].as<std::bitset<9>>()),
+                                        ErrorMsg::BAD_CONVERSION);
+}
+
+TEST(NodeTest, StdBitsetWrongRepresentation) {
+  const std::string representation =
+      "0011"
+      "1000"
+      "1101"
+      "x000";  // 16 bits, note the wrong character 'x' here
+  Node node;
+  node["not_really_bits"] = representation;
+  EXPECT_THROW_REPRESENTATION_EXCEPTION(
+      (node["not_really_bits"].as<std::bitset<16>>()),
+      ErrorMsg::BAD_CONVERSION);
+}
+
+TEST(NodeTest, ConversionsBetweenSequences) {
+  const std::vector<int> primesAsVec{2, 3, 5, 7, 11, 13};
+
+  const auto node = Node(primesAsVec);
+
+  // should succeed -- yaml-cpp doesn't care what kind of underlying sequence
+  // type it was.
+  const auto primesAsDeque = node.as<std::deque<int>>();
```

Comment:
I didn't use testing::ElementsAreArray because `std::array` screwed up that function. Anyway, fixed in c34a8bf